### PR TITLE
Tests: skip db concurrent Read/Write tests if test.short is set

### PR DESCRIPTION
**Pull request by algoidan**
## Summary

Some of the `dbutil` tests run for a long time and simulate some specific edge cases on the databases infrastructure.  it is safe to run those tests only on nightly jobs.

### CODE REVIEW

File: util/db/dbutil_test.go
Comment (michaeldiamant) [inline util/db/dbutil_test.go:242]: @algoidan Optional - If you feel it's worthwhile, I propose adding an explanation in `t.Skip("...")` (or inline comment) explaining why we think it's reasonable to skip during short test runs.  I think it will help future maintainers judge how to treat the test.

If you prefer as is, feel welcomed to resolve as is.

Aside:  I'm holding off on providing approval because I'm far enough removed from this part of the repo that I don't feel comfortable approving the change to nightly builds.

**Review by algorandskiy — COMMENTED** [review]
+1 to Michael's comment, it is better to have the skipping reason right there

File: util/db/dbutil_test.go
Comment (algoidan) [reply util/db/dbutil_test.go]: I added a comment for this. Please let me know if that is a reasonable explanation.

Comment (michaeldiamant) [reply util/db/dbutil_test.go]: @algoidan Resolving - Thanks for adding a note.

Keep in mind, I _won't_ add approval because I don't consider myself knowledgeable enough to make a call here.

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**
